src/harness/testcases/WINNF_FT_S_FAD_testcase.py

```diff
@@ -0,0 +1,284 @@
+              'fccId': device['fccId'],
+              'fccMaxEirp': 47
+          })
+          self._sas_admin.InjectUserId({'userId': device['userId']})
+          
+        # Pre-load conditional registration data for N3 CBSDs.
+        self._sas_admin.PreloadRegistrationData(
+            config['conditionalRegistrationData'])
+        
+        # Register N1 CBSDs.
+        request = {'registrationRequest': config['registrationRequests']}
+        responses = self._sas.Registration(request)['registrationResponse']
+        # Check registration responses and get cbsd Id
+        grants = config['grants']
+        self.assertEqual(len(responses), len(config['registrationRequests']))
+        for index, response in enumerate(responses):
+            self.assertEqual(response['response']['responseCode'], 0)
+            grants[index]['cbsdId'] = response['cbsdId']
+        # send grant request with N1 grants
+        del responses
+        grant_responses = self._sas.Grant({'grantRequest': grants})['grantResponse']
+        # check grant response
+        self.assertEqual(len(grant_responses), len(config['grants']))
+        for grant_response in grant_responses:
+            self.assertEqual(grant_response['response']['responseCode'], 0)       
+        # inject N2 PPAs and PALs
+        ppa_ids = []
+        for index, ppa in enumerate(config['ppas']):
+            self._sas_admin.InjectPalDatabaseRecord(config['pals'][index])
+            ppa_ids.append(self._sas_admin.InjectZoneData({'record': ppa}))          
+        # inject N3 Esc sensor
+        for esc_sensor in config['esc_sensors']:
+            self._sas_admin.InjectEscSensorDataRecord({'record': esc_sensor})   
+            
+        # STEP 3
+        self._sas_admin.TriggerFullActivityDump()
+        # STEP 4
+        response = self._sas.GetFullActivityDump()
+        # STEP 5
+        # check dump message format
+        self.assertContainsRequiredFields("FullActivityDump.schema.json", response)
+        # an array for each record type
+        cbsd_dump_data = []
+        ppa_dump_data = []
+        esc_sensor_dump_data = []
+        # STEP 5
+        self.assertContainsRequiredFields("FullActivityDump.schema.json", response)
+        
+        # step 6 and check   
+        # download dump files and fill corresponding arrays
+        for dump_file in response['files']:
+            self.assertContainsRequiredFields("ActivityDumpFile.schema.json",
+                                               dump_file[0])
+            if dump_file['recordType'] ==  'cbsd':
+                cbsd_dump_data.append(self._sas.DownloadFile(dump_file['url'])['recordData'])   
+            elif dump_file['recordType'] ==  'esc_sensor':
+                esc_sensor_dump_data.append(self._sas.DownloadFile(dump_file['url'])['recordData'])
+            elif dump_file['recordType'] ==  'zone':
+                ppa_dump_data.append(self._sas.DownloadFile(dump_file['url'])['recordData'])
+        
```

Comment:
I think the dump can contain an empty coordination events dump also, right?
so I added one verification that the other files are only of coordination events record type,

Reply:
My understanding is that CoordinationEvents are not used at this time (in part because no one has defined exactly what information to include or how a SAS should respond). If a SAS has recently been Reset(), I'm not sure why it would have a CoordinationEvent record. Do you have an example in mind?

Reply:
I do not have strong opinion on this CoordinationEvents. However, for making it simpler and cleaner, I would suggest not including it.

Reply:
As I remember, the SAS can generate a dump file (it will be an empty file if no corresponding data in SAS) for all types of  records in SAS-SAS sepc, isn't that correct.

Reply:
I guess it's not forbidden (the spec agrees with you), it's just not something I had even considered doing. I'm OK to keep the current implementation as long as you add a comment to explain why you chose it. Otherwise someone -- probably me ;) -- will be confused in a few months when they've forgotten (or not read) this conversation.

Are there any other record types that may be included in the FAD, or is it just CoordinationEvent?